cmake/version.cmake

```diff
@@ -5,14 +5,19 @@ if(NOT REVISION)
   string(SUBSTRING "${GIT_SHA1}" 0 8 GIT_SHA1_SHORT)
   git_describe(GIT_DESCRIPTION)
   git_describe(GIT_TAG_LAST "--abbrev=0" "--tags")
-  git_get_exact_tag(GIT_TAG_EXACT)
-  string(SUBSTRING "${GIT_TAG_LAST}+${GIT_SHA1_SHORT}" 1 -1 REVISION)
-  if(GIT_TAG_EXACT)
-    string(SUBSTRING "${GIT_TAG_EXACT}" 1 -1 REVISION)
-    message(STATUS "Currently on Git tag ${GIT_TAG_EXACT}")
-  else ()
-    message(STATUS "Latest recognized Git tag is ${GIT_TAG_LAST}")
-    set(GIT_TAG_EXACT "")
+  if(GIT_TAG_LAST MATCHES "-NOTFOUND")
+    set(REVISION "0.0.0+${GIT_SHA1_SHORT}")
+    message(STATUS "Using PEP 440 compliant version for shallow clone")
```

Comment:
It would be helpful to explicitly log tag lookup failures to clarify fallback reasons for easier debugging.
e.g., message(WARNING "No valid tag reachable, using fallback 0.0.0+${GIT_SHA1_SHORT}")